pkg/api/types.go

```diff
@@ -180,6 +180,9 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// ISCSIDiskVolumeSource represents an ISCSI Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	ISCSIDisk *ISCSIDiskVolumeSource `json:"iscsiDisk"`
```

Comment:
in terms of API and how it looks, less is more, I think.  I don't think "Disk" adds anything here.  I can be swayed, but I think

```json
"iscsi": { "targetPortal": "1.2.3.4:5678", ...}
``` 

is marginally cleaner than

```json
"iscsiDisk": { "targetPortal": "1.2.3.4:5678", ...}
```